### ISSUE TITLE
Add ability to disallow adding accio script to targets

### ISSUE THREAD

**Issue by insightmind** [JamitLabs/Accio#25]
The Accio script causes several "not permitted" and "directory not found" issues when building a project with extensions, e.g. NotificationContentExtensions.

When adding extensions to an Xcode project Xcode creates a new target for each extension. In some cases, we need to access some functionality of the app, e.g. accessing a Realm Database within a Notification Extension requires to include the Realm Framework to the Extension Target.
By doing this and running `accio install` Accio automatically adds the Accio script to the target. This will cause problems while building the project because several scripts are executed at build time. Therefore I propose to add a flag or possibility to disallow adding the Accio script to a specific target.

**Comment by Jeehut** [comment]
@insightmind Thank you for reporting the issue. One question: If we could automatically detect if a target is an extension and don‘t create the script there, would that also fix the issue or would there still be situations where such an option would still be useful?

Having no option but everything still working correctly would be my preferred way to go, unless there‘s evidence that we probably won‘t be able to automatically cover all cases.

**Comment by insightmind** [comment]
# Using `WRAPPER_EXTENSION`
![Screen Shot 2019-04-18 at 9 49 02 AM](https://user-images.githubusercontent.com/24965150/56345103-3f4a6a00-61bf-11e9-9fb3-4132a1c94e1d.png)

There is a property defined in the Build Settings called` WRAPPER_EXTENSION`, which defines the type of the target, e.g for an `app`, for a TestTarget `xctest` (#23), or for an AppExtension: `appex`.

Documentation:
> The extension used for product wrappers, which has a default value based on the product type.

However, the current documentation at https://help.apple.com/xcode/mac/10.2/#/itcaec37c2a6 does not state which values this property can take. This could result in problems in future OS version and additionally, we might never be able to handle all cases.

# Alternatives:
## SwiftLint Style
_SwiftLint_ uses inline comments to allow and disallow certain rules at the specified code line.
We could adapt this idea by providing custom inline comments for the `Package.swift` file.

e.g.

```swift
[...]
.target( // accio:noScript
   name: "StopWatchNotificationContentExtension",
   dependencies: [
      "RealmSwift",
      "SwiftyUserDefaults",
   ],
   path: "StopWatchNotificationContentExtension"
),
[...]
```

or the logical negation:

```swift
[...]
.target( // accio:script
   name: "App",
   dependencies: [
      "RealmSwift",
      "SwiftyUserDefaults",
   ],
   path: "App"
),
[...]
```

I would recommend this way as it is an easy approach which can be shared across team members and it also embraces an opportunity to add upcoming features for Accio in the same way.

## Accio command line
We could also use a command line flag:

`accio install -noScript 'StopWatchNotificationContentExtension'`

or:

`accio install -script 'App'`

However, this may causes problems when working in a team as every developer needs to remember which targets should allow a script or disallow it.

**Comment by Jeehut** [comment]
@insightmind Thank you for the detailed investigation. I'm clearly voting for going with the first approach and will implement it alongside the fix for #23. Even when Apple had the possible values documented, this still wouldn't mean they could never change in future versions. But I don't think they will change often and if they do, we will recognize that pretty quickly and can fix it easily.

**Comment by Jeehut** [comment]
This was implemented in bf0dfaaac5e7425ca41a6bbe07571e46b52cf638.
